py/nlrthumb.c

```diff
@@ -44,7 +44,19 @@ __attribute__((naked)) unsigned int nlr_push(nlr_buf_t *nlr) {
     "str    r6, [r0, #20]       \n" // store r6 into nlr_buf
     "str    r7, [r0, #24]       \n" // store r7 into nlr_buf
 
-#if defined(__ARM_ARCH_6M__)
```

Comment:
I think it would be equivalent, and a much smaller patch, to change this line to `#if !defined(__thumb2__)` (also other 3 sections below).

Reply:
We can certainly do that if keeping the diff to a minimum is important. I like to avoid the `!` in the `if` clause when there is also an `else. I think it makes the code a little easier to follow.

Reply:
> We can certainly do that if keeping the diff to a minimum is important.

It is somewhat important.  But also the reason to keep the current ordering is because then the "less capable" CPU has its code first, followed by the "more capable" CPU.

(I don't think it'll work to use `defined(__thumb__)` because that's defined for both Thumb and Thumb 2 archs.}

Reply:
OK. I'll fix it up as suggested.